trucks.js

```diff
@@ -248,9 +255,28 @@ var foodTrucks = [
   }
 ];
 
-/* 
- * return an object from the module with a single method on it: filterByDay
- * that method should take a single parameter that represents the day to filter on
- * use the built-in filter() method to return all trucks that have the day in their
- * schedule
- */
\ No newline at end of file
+//function returns true if the currently checked truck is available
+//on the target day
+function filterBySchedule(truck, day) {
+  if (truck.schedule.indexOf(day) !== -1) {
+    return true;
+  } else {
+    return false;
+  }
+}
+ 
+//filters a list of trucks by their schedule and returns an array of trucks
+//that are available on a certain day
+var filterTrucksByDay = function(day){
+  var filteredTruckArray = foodTrucks.filter(function(truck){
```

Comment:
Better to return the this instead of extracting the names. There may be other pieces of information desired by the consumer of the filterTrucksByDay method.